### PR TITLE
Switch to adaptor signatures

### PR DESCRIPTION
**Pull request by Tibo-lg**

### CODE REVIEW

File: src/cfddlc_transactions.cpp [hunk 1]
```diff
@@ -763,23 +471,53 @@ uint32_t DlcManager::GetTotalInputVSize(const std::vector<TxIn>& inputs) {
...
+    }
+    fund_weight =
+        36 + option_dest.GetLockingScript().GetData().GetDataSize() * 4;
+  }
+  auto fund_fee = ceil(fund_weight / 4) * fee_rate;
+  double cet_weight = (CET_BASE_WEIGHT / 2 +
+                       params.final_script_pubkey.GetData().GetDataSize() * 4);
+  auto cet_fee = ceil(cet_weight / 4) * fee_rate;
+  auto fund_out = params.collateral + fund_fee + cet_fee;
+  if (params.input_amount < fund_out) {
```
Comment (k-matsuzawa) [inline src/cfddlc_transactions.cpp]: I think you should also consider option_premium in this conditional statement.
```suggestion
  if (params.input_amount < (fund_out + option_premium)) {
```

File: src/cfddlc_transactions.cpp [hunk 2]
```diff
@@ -763,23 +471,53 @@ uint32_t DlcManager::GetTotalInputVSize(const std::vector<TxIn>& inputs) {
...
+      (FUND_TX_BASE_WEIGHT / 2 + inputs_size + change_size * 4 + 36);
+  if (option_premium.GetSatoshiValue() > 0) {
+    if (option_dest.GetAddress() == "") {
+      throw CfdException(
+          CfdError::kCfdIllegalArgumentError,
+          "An destination address for the premium is required when the option "
+          "premium amount is greater than zero.");
+    }
+    fund_weight =
+        36 + option_dest.GetLockingScript().GetData().GetDataSize() * 4;
```
Comment (k-matsuzawa) [inline src/cfddlc_transactions.cpp]: What you want to do here is to add the size of the added output to fund_weight.
```suggestion
    fund_weight +=
        36 + option_dest.GetLockingScript().GetData().GetDataSize() * 4;
```

**Review by k-matsuzawa — COMMENTED** [review]
@Tibo-lg 
I reviewed it, but basically there is no problem!
There was a careless mistake when using `option_premium` with the `GetChangeOutputAndFees` function, so I added a comment. Please check. :pray:

File: src/cfddlc_transactions.cpp (hunk 1 above)
Comment (Tibo-lg) [reply src/cfddlc_transactions.cpp]: Thanks! Fixed

File: src/cfddlc_transactions.cpp (hunk 2 above)
Comment (Tibo-lg) [reply src/cfddlc_transactions.cpp]: Thanks!

**Pull request merged**